[lakshm1i] iP

lakshm1i opened the pull request.

Commit:
Started on iP and did level 0

I renamed the chatbot from Duke to Yale. I created two methods: greetUser to greet the user and exitChat to display the exit message.

Commit:
Updated code according to instructions.

Created a echo method. It uses scanner to read a line of input from the keyboard and if it is not bye (ignoring case), it will echo back the input. If input is bye, will exitchat.

Commit:
Updated code according to instructions

Upgraded echo method to list the task or add the task depending on input. addTask will add input into array of size 100 and increase size of array accordingly. listTask will list down task number and task name.

terryasdf commented on src/main/java/Yale.java:
```diff
@@ -0,0 +1,51 @@
+import java.util.Scanner;
+
+public class Yale {
+    private static String[] tasks = new String[100]; //Array to store tasks
```
You could use a constant to store the 100 value.

Commit:
Added ability to mark and unmark tasks as done.

First, instead of having all methods (greet, exit,echo) in one class, i seperated them into different class (taskmanager, ui). Then, i created task class that stores if its done and description. when user says mark a certain task, yale will put X in brackets and print out the task accordingly. when commanded list, for each task, will show in brackets if its done or not.

Commit:
Added support to track type of tasks such as todo, deadline and event.

created 3 more classes for todo, deadline and event. each were inherited from task class and each overrides to string accordingly. for deadline and event, split function was used to split the sentence into arrays of description of tasks and the time information. during "list", will list out the task type, if its marked/unmarked, description and the timing if applicable. Also changed the if-else structure of runChat slightly.

terryasdf reviewed (commented):
Well done!

Commit:
A-CodingStandard

Changed the naming convention of some variable and method names to follow the Java’s camelCase convention. For example, in to inputScanner, line to userInput, task to taskList, index to taskIndex, listTask to displayTaskList, runChat to hancleUserInput

Commit:
I improved the quality of code.

handleUserInput was quite long, so I refactored into smaller methods (processUserInput, processMarkTask, processUnmarkTask, etc.) to improve readability and maintainability.
instead of hardcoding array size 100, i put it in a variable MAX_TASKS to eliminate magic numbers.
added scenarios for invalid inputs and commands to improve readability.
added checks for empty descriptions and invalid task indices to prevent runtime errors.
tried to ensure each method operates at a single level of abstraction.

Commit:
handling errors

code will print error message for the various error scenarios such as when todo is empty, when deadline and event is in wrong format (dont have slashes), when the input is none of the existing features, when index of task to be marked/unmarked is not valid, when adding tasks after 100 tasks have been added, when trying to list when list is empty.

Commit:
used exceptions to handle errors

defined a class YaleException to represent exceptions specific to Yale. I added throwing exception to when todo is empty, event/deadline is in invalid format and when users input is invalid command.

Alfred-Goh02 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,32 @@
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
```
use K&R style brackets for if else statement

Alfred-Goh02 commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,51 @@
+    private static final int MAX_TASKS = 100;
+    private final Task[] taskList = new Task[MAX_TASKS]; //Array to store tasks
+    private int taskCounter = 0; //Track number of tasks stored
+
```
Consider separating printing statements and adding it into your UI class.

Alfred-Goh02 commented on src/main/java/Yale.java:
```diff
@@ -0,0 +1,88 @@
+    private static void processUserInput(String userInput, TaskManager taskManager) throws YaleException {
+        if (userInput.equalsIgnoreCase("list")) {
+            taskManager.displayTaskList();
+        } else if (userInput.startsWith("mark ")) {
```
All these command strings like list, mark etc. should be declared as a constant at the start of the file. Avoid magic literals.

Commit:
I replaced the fixed array with dynamic array using ArrayList<Task>. I have also created a jar file.

I also changed the other methods like addTask, displayTask, mark and unmarkTask accordingly as there is no more MAX_TASKS and taskcounter.

Commit:
created method to delete tasks from the list

in taskmanager, created a code for deleteTask, which removes a valid index list. in main code, ensured that if command begins with "delete", the command will be processed as deletetask

Commit:
level 7, save

created a new file storage to load and save tasks. also changed event task deadline to implement getters. used dynamic arrays in taskmanager.

Commit:
Merge branch 'branch-Level-6'

* branch-Level-6:
  created method to delete tasks from the list
  I replaced the fixed array with dynamic array using ArrayList<Task>. I have also created a jar file.

Commit:
level-6 save

Commit:
Merge branch 'branch-Level-7'

* branch-Level-7:
  level 7, save

Commit:
resolved merge conflicts

delete function was missing, so added it back

Commit:
Improved code

According to feedback from TA.

Commit:
Made the code more OOP.

- Extracted functionalities into separate classes for better organization and readability:
  - Ui: Handles user interactions and displays relevant messages.
  - Storage: Manages task file loading and saving.
  - Parser: Processes user commands and executes corresponding actions.
  - TaskList: Manages task-related operations like adding, deleting, and marking tasks.

Commit:
implemented level 9, find function

find method is added in tasklist, parser handles find command and ui shows the matching tasks

Commit:
Added JavaDoc comments to the code.

Add header comments to the non-private classes/methods in all files

Commit:
Revert "Added JavaDoc comments to the code."

This reverts commit ca74f9040bb5ae6970a3f533d790597863e5b2f1.

Commit:
Added JavaDoc comments to the code.

Added header comments to the non-private classes/methods in all classes.

lakshm1i committed:
Merge pull request #1 from lakshm1i/branch-Level-9

Branch level 9

lakshm1i committed:
Merge branch 'master' into branch-A-JavaDoc

lakshm1i committed:
Merge pull request #2 from lakshm1i/branch-A-JavaDoc

Added JavaDoc comments to the code.

Commit:
Added a User Guide for Yale

by stating description, command, example and expected output for each feature.
